common/src/Model/EntityNodeIndex.cpp

```diff
@@ -128,16 +128,15 @@ namespace TrenchBroom {
         }
 
         std::vector<EntityNodeBase*> EntityNodeIndex::findEntityNodes(const EntityNodeIndexQuery& keyQuery, const std::string& value) const {
-            const std::set<EntityNodeBase*> nameResult = keyQuery.execute(*m_keyIndex);
-
+            // first, find Nodes which have `value` as the value for any key
             std::set<EntityNodeBase*> valueResult;
             m_valueIndex->find_matches(value, std::inserter(valueResult, std::end(valueResult)));
-            if (nameResult.empty() || valueResult.empty()) {
+            if (valueResult.empty()) {
```

Comment:
While you're here, it might also be faster to just use a vector to collect the result, and then call `kdl::sort_and_remove_duplicates` on it than using `std::set`. I tend to replace most uses of `std::set` with that pattern these days.

Reply:
Agreed, will do.